lib/KImg/__tests__/KImg.spec.js

```diff
@@ -1,92 +1,104 @@
-  });
 
-  it(`throws an error when 'aspectRatio' has an invalid format`, () => {
-    const error = jest.fn();
-    makeWrapper({
-      propsData: { src: '/le-logo.svg', altText: 'LE logo', aspectRatio: '16/9' },
-      listeners: { error },
+      it(`sets 'alt' attribute to an empty string`, () => {
+        const wrapper = makeWrapper({
+          propsData: { src: '/le-logo.svg', altText: undefined, isDecorative: true },
+        });
+        expect(wrapper.exists()).toBe(true);
+        expect(wrapper.find('img').attributes('alt')).toBe('');
+      });
     });
-    expect(error).toHaveBeenCalled();
-    expect(error.mock.calls[0][0]).toBeInstanceOf(Error);
-    expect(error.mock.calls[0][0].message).toBe('Invalid aspect ratio provided: 16/9');
-  });
 
-  it(`doesn't throw an error when 'aspectRatio' has a valid format`, () => {
-    const error = jest.fn();
-    makeWrapper({
-      propsData: { src: '/le-logo.svg', altText: 'LE logo', aspectRatio: '16:9' },
-      listeners: { error },
+    it(`throws an error when 'aspectRatio' has an invalid format`, () => {
+      const error = jest.fn();
+      makeWrapper({
+        propsData: { src: '/le-logo.svg', altText: 'LE logo', aspectRatio: '16/9' },
+        listeners: { error },
+      });
+      expect(error).toHaveBeenCalled();
+      expect(error.mock.calls[0][0]).toBeInstanceOf(Error);
+      expect(error.mock.calls[0][0].message).toBe('Invalid aspect ratio provided: 16/9');
     });
-    expect(error).not.toHaveBeenCalled();
-  });
 
-  it(`emits an 'error' event when there is an error in loading the image`, async () => {
-    const error = jest.fn();
-    const wrapper = makeWrapper({
-      propsData: { src: '/le-logo.svg', altText: 'LE logo' },
-      listeners: { error },
+    it(`doesn't throw an error when 'aspectRatio' has a valid format`, () => {
+      const error = jest.fn();
+      makeWrapper({
+        propsData: { src: '/le-logo.svg', altText: 'LE logo', aspectRatio: '16:9' },
+        listeners: { error },
+      });
+      expect(error).not.toHaveBeenCalled();
     });
 
-    // Manually trigger the onError method to simulate the image load failure
-    const e = new Event('error');
-    wrapper.vm.onError(e);
+    it(`emits an 'error' event when there is an error in loading the image`, async () => {
+      const error = jest.fn();
+      const wrapper = makeWrapper({
+        propsData: { src: '/le-logo.svg', altText: 'LE logo' },
+        listeners: { error },
+      });
+
+      // Manually trigger the onError method to simulate the image load failure
+      const e = new Event('error');
+      wrapper.vm.onError(e);
 
-    expect(error).toHaveBeenCalled();
-    expect(error.mock.calls[0][0]).toBeInstanceOf(Event);
-    expect(error.mock.calls[0][0]).toEqual(e);
-  });
+      expect(error).toHaveBeenCalled();
+      expect(error.mock.calls[0][0]).toBeInstanceOf(Event);
+      expect(error.mock.calls[0][0]).toEqual(e);
+    });
+  }
+  if (takeScreenshot()) {
```

Comment:
Hmm I think that we sould prefer this to be more transparent for developers. I think what @bjester was suggesting was to run the unit tests and that we could run an "snapshot" function that will only be excecuted if we were on visual tests mode. Something like
```js
    it(`renders without any errors when a valid 'src' and 'altText' are provided`, () => {
      const error = jest.fn();
      const wrapper = makeWrapper({ // This method should also allow to render the component dynamically on the dev server, so we can take the snapshot later.
        propsData: { src: '/le-logo.svg', altText: 'LE logo' },
        listeners: { error },
      });

      takeSnapshot(); // This will call an util function that will take the percy snapshot if we are running visual tests. If we are running unit tests we will just ignore it.

      expect(wrapper.exists()).toBe(true);
      expect(error).not.toHaveBeenCalled();

      const img = wrapper.find('img');
      expect(img.exists()).toBe(true);
      expect(img.attributes('src')).toBe('/le-logo.svg');
      expect(img.attributes('alt')).toBe('LE logo');
    });
```

I think that was what @bjester was talking about, but you can correct me if I misunderstood :open_hands:

Reply:
yes @AlexVelezLl, I think that would be ideal, as it is simpler and minimizes the implementation in each test case required to take a screenshot.

The main reason I brought up the [testing library](https://testing-library.com/docs/) was because of the differences in how the test might need to be set up relative to unit testing. I supposed the testing library could be leveraged to achieve the simpler implementation we ideally seek, because the API has similarities between the different framework-specific packages. 

I can see there would likely be some challenges in moving forward with the ideal approach. I also think it's where some careful software engineering can be employed to put the pieces together, and where some real value will be added in this project to keep maintenance and overhead low.

Reply:
We decided to keep the current change for now to avoid blocking the PR, since more decisions will be needed. For next steps we will first determine how we will be rendering the components in the dev server, and after we can continue with how we can abstract this logic for developers.